Create and use DynamicTypes for check in DispatchKeyExtractor::makeBitsetForDispatchArgs

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #151850
* #151849
* #151810
* #151807
* #151806
* #151805
* #151804
* #151803
* __->__ #151802
* #151801

On mobile, many but not all things in the JIT type subsystem start using DynamicType. Not using DynamicType  was imposing a startup time cost here, as explained in the comment.

Differential Revision: [D73129442](https://our.internmc.facebook.com/intern/diff/D73129442/)

swolchok committed:
Create and use DynamicTypes for check in DispatchKeyExtractor::makeBitsetForDispatchArgs

On mobile, many but not all things in the JIT type subsystem start using DynamicType. Not using DynamicType  was imposing a startup time cost here, as explained in the comment.

Differential Revision: [D73129442](https://our.internmc.facebook.com/intern/diff/D73129442/)

[ghstack-poisoned]

swolchok committed:
Update on "Create and use DynamicTypes for check in DispatchKeyExtractor::makeBitsetForDispatchArgs"


On mobile, many but not all things in the JIT type subsystem start using DynamicType. Not using DynamicType  was imposing a startup time cost here, as explained in the comment.

Differential Revision: [D73129442](https://our.internmc.facebook.com/intern/diff/D73129442/)

[ghstack-poisoned]

swolchok committed:
Update on "Create and use DynamicTypes for check in DispatchKeyExtractor::makeBitsetForDispatchArgs"


On mobile, many but not all things in the JIT type subsystem start using DynamicType. Not using DynamicType  was imposing a startup time cost here, as explained in the comment.

Differential Revision: [D73129442](https://our.internmc.facebook.com/intern/diff/D73129442/)

[ghstack-poisoned]

swolchok committed:
rebase and minor fixes as requested on "Create and use DynamicTypes for check in DispatchKeyExtractor::makeBitsetForDispatchArgs"


On mobile, many but not all things in the JIT type subsystem start using DynamicType. Not using DynamicType  was imposing a startup time cost here, as explained in the comment.

Differential Revision: [D73129442](https://our.internmc.facebook.com/intern/diff/D73129442/)

[ghstack-poisoned]

swolchok commented:
ping reviewers @malfet @seemthere @atalman @izaitsevfb

malfet reviewed (approved):
Somebody more familiar with what is the current use/scope of C10_MOBILE should review that code part, but non-mobile part of change LGTM

swolchok committed:
Update on "Create and use DynamicTypes for check in DispatchKeyExtractor::makeBitsetForDispatchArgs"


On mobile, many but not all things in the JIT type subsystem start using DynamicType. Not using DynamicType  was imposing a startup time cost here, as explained in the comment.

Differential Revision: [D73129442](https://our.internmc.facebook.com/intern/diff/D73129442/)

[ghstack-poisoned]
